sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,436 @@
+from sympy.solvers import linsolve
+from sympy.printing import sstr
+from sympy.functions import SingularityFunction
+from sympy.core import sympify
+from sympy.integrals import integrate
+
+
+class Beam(object):
+    """
+    A Beam is a structural element that is capable of withstanding load
+    primarily by resisting against bending. Beams are characterized by
+    their cross sectional profile(Second moment of area), their length
+    and their material.
+
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import Symbol, Piecewise
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> b = Beam(4, E, I)
+    >>> b.apply_load(order=-1, value=-3, start=0)
+    >>> b.apply_load(order=-1, value=-9, start=4)
+    >>> b.apply_load(order=0, value=6, start=2)
+    >>> b.apply_boundary_conditions(deflection=[(4, 0)])
+    >>> b.boundary_conditions
+    {'deflection': [(4, 0)], 'moment': [], 'slope': []}
+    >>> b.load_distribution()
+    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
+    >>> b.shear_force()
+    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
+    >>> b.bending_moment()
+    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*SingularityFunction(x, 4, 1)
+    >>> b.slope()
+    (3*SingularityFunction(x, 0, 2)/2 - SingularityFunction(x, 2, 3) + 9*SingularityFunction(x, 4, 2)/2 - 7)/(E*I)
+    >>> b.deflection()
+    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
+    >>> b.deflection().rewrite(Piecewise)
+    (-7*x + Piecewise((x**3, x > 0), (0, True))/2
+          + 3*Piecewise(((x - 4)**3, x - 4 > 0), (0, True))/2
+          - Piecewise(((x - 2)**4, x - 2 > 0), (0, True))/4)/(E*I)
+    """
+
+    def __init__(self, length, elastic_modulus, second_moment, variable=Symbol('x')):
+        """Initializes the class.
+
+        Parameters
+        ==========
+        length : Sympifyable
+            A SymPy expression representing the Beam's length.
+        elastic_modulus : Sympifyable
+            A SymPy expression representing the Beam's Modulus of Elasticity.
+            It is a measure of the stiffness of the Beam material.
+        second_moment : Sympifyable
+            A SymPy expression representing the Beam's Second moment of area.
+            It is a geometrical property of an area which reflects how its
+            points are distributed with respect to its neutral axis.
+        variable : Symbol
+            A Symbol object that will be used as the variable along the beam
+            while representing the load, shear, moment, slope and deflection
+            curve. By default, it is set to ``Symbol('x')``.
+        """
+        self._length = length
+        self._elastic_modulus = elastic_modulus
+        self._second_moment = second_moment
```

Comment:
The arguments will be simpified if you use the notation I list above because you have the sympfy calls in the setters, no?